### PR TITLE
Worldgen rocks but no resources

### PR DESCRIPTION
**Pull request by dsecrieru**
Skipped resources since 0.3 has its own system for it.

PR also has a fix to the fix for Linux, which introduced a bug for Windows :).

### CODE REVIEW

**Comment by dsecrieru** [comment]
One more thing, if you run it, you can change between the map modes with 'e' for elevation and 'r' for rocks in the worldgen scene.

File: sote/libsote/inorganic_stone_layer.lua [hunk 1]
```diff
@@ -0,0 +1,211 @@
+local isl = {}
```
Comment (Calandiel) [inline sote/libsote/inorganic_stone_layer.lua]: This belongs in raws, not in libsote, imho

File: sote/libsote/rock_layers.lua [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+for _, value in pairs(rock_types) do
+    rock_layers[value] = {}
+end
+
+require("libsote.stone_layers_loader").load()
+local stone_layers = require("libsote.stone_layer").get_layers_by_id()
+
+local function assign_igneous_extrusive(stone_layer)
+    if stone_layer.acidity < 0.33000001311302185 then
```
Comment (Calandiel) [inline sote/libsote/rock_layers.lua]: Can we fix these constants to be more readable? Its a rounding error from original 0.33, wont matter in practice unless you need perfect reproducibility for debugging purposes

Comment (dsecrieru) [reply sote/libsote/rock_layers.lua]: fixed

File: sote/libsote/inorganic_stone_layer.lua (hunk 1 above)
Comment (dsecrieru) [reply sote/libsote/inorganic_stone_layer.lua]: I've also considered this, in raws we already have bedrocks, but they lack some of the attributes the materials from C# have and need, so I decided not to extend bedrocks raws, nor to put the stone materials together with bedrocks, which might cause confusion.

Comment (Calandiel) [reply sote/libsote/inorganic_stone_layer.lua]: They are the exact same thing, if we are porting world gen they will need this new data

Comment (dsecrieru) [reply sote/libsote/inorganic_stone_layer.lua]: Makes sense from that perspective. I will see about merging them.

Comment (dsecrieru) [reply sote/libsote/inorganic_stone_layer.lua]: Just an observation, ROCK_SALT, DOLOMITE and OBSIDIAN are missing from bedrocks.

Comment (Calandiel) [reply sote/libsote/inorganic_stone_layer.lua]: I think I removed them as they weren't implemented well enough. We can add them back for the purpose of the port @dsecrieru

Comment (dsecrieru) [reply sote/libsote/inorganic_stone_layer.lua]: I can add them, but I don't know what values to input for sand, clay, silt, organics, minerals and weathering, so I'll just default them to 0 for now.
Also note that there is an "obsidian" resource, based on "rhyolite" bedrock. No idea if that has any relevance or not.

Comment (dsecrieru) [reply sote/libsote/inorganic_stone_layer.lua]: Looks like "dolomite" is present in bedrocks as "dolostone"

Comment (Calandiel) [reply sote/libsote/inorganic_stone_layer.lua]: Quite possible, it's been a while, I don't remember. Tell me when you think it's ready for merging again @dsecrieru ^-^

**Comment by dsecrieru** [comment]
Ok, ready to merge @Calandiel! :)

**Comment by Calandiel** [comment]
Thanks!

**Pull request merged**